examples/decomposition/plot_faces_decomposition.py

```diff
@@ -122,7 +123,8 @@ def plot_gallery(title, images, n_col=n_col, n_row=n_row):
         components_ = estimator.cluster_centers_
     else:
         components_ = estimator.components_
-    if hasattr(estimator, 'noise_variance_'):
+    if (hasattr(estimator, 'noise_variance_') and
+            estimator.noise_variance_.shape != ()):
```

Comment:
thanks @lesteve for the clarification. Makes more sense to me now.
I just pushed an update containing comments.